### PR TITLE
Remove unused cimports

### PR DESCRIPTION
**Pull request by jbrockmendel**
This helps a little bit with the complexity of the libs interdependencies.

Make properties.pyx its own module, since we will end up needing it to be separate when `tseries.offsets` gets moved into cython.

Remove the long-defunct _libs/src/offsets.pyx.

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
you are doing quite a lot of things here. specifically don't include the property changes here.

**Comment by jbrockmendel** [comment]
properties separated out into #17590

File: pandas/_libs/lib.pyx [hunk 1]
```diff
@@ -67,6 +67,7 @@ import tslib
 from tslib import NaT, Timestamp, Timedelta
 import interval
 from interval import Interval
+from properties import AxisProperty, cache_readonly
```
Comment (jreback) [inline pandas/_libs/lib.pyx]: are you adding these back for a reason?

File: pandas/_libs/src/offsets.pyx [hunk 2]
```diff
@@ -1,367 +0,0 @@
...
-    if dow > 4:                         # sat or sun?
-        if bkwd:                        # roll back 1 or 2 days
-            return (4 - dow)
-        else:                           # roll forward 2 or 1 days
-            return (7 - dow)
-    return 0
-
-cdef int64_t us_in_day = conversion_factor(r_microsecond, r_day)
-
-cdef class _Offset:
```
Comment (jreback) [inline pandas/_libs/src/offsets.pyx]: anything useful from this file?

File: setup.py [hunk 3]
```diff
@@ -476,6 +476,7 @@ def pxd(name):
 libraries = ['m'] if not is_platform_windows() else []
 
 ext_data = {
+    '_libs.properties': {'pyxfile': '_libs/properties', 'include': []},
```
Comment (jreback) [inline setup.py]: duplicated

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/lib.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/lib.pyx]: Looks like a rebase screwup.  Will fix.

File: pandas/_libs/src/offsets.pyx (hunk 2 above)
Comment (jbrockmendel) [reply pandas/_libs/src/offsets.pyx]: I looked through it and didn't find anything worth keeping.  I've got a cython version of `tseries.offsets` in a branch that is a few PRs away from being ready to push upstream.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**